src/Views/OperatingSystemView.vala

```diff
@@ -53,11 +56,33 @@ public class About.OperatingSystemView : Gtk.Grid {
             logo_icon_name = "distributor-logo";
         }
 
-        var logo = new Gtk.Image () {
-            halign = Gtk.Align.END,
-            icon_name = logo_icon_name,
-            pixel_size = 128
+        var logo = new Hdy.Avatar (128, "", false) {
+            // In case the wallpaper can't be loaded, we don't want an icon or text
+            icon_name = "invalid-icon-name",
+            // We need this for the shadow to not get clipped by Gtk.Overlay
+            margin = 6
         };
+        logo.set_image_load_func ((size) => {
+            try {
+                return new Gdk.Pixbuf.from_file_at_scale ("/usr/share/backgrounds/elementaryos-default", -1, size, true);
+            } catch (Error e) {
+                critical (e.message);
+            }
+        });
+        logo.get_style_context ().add_provider (style_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
+
+        var icon = new Gtk.Image () {
+            icon_name = logo_icon_name + "-symbolic",
+            pixel_size = 122
```

Comment:
Might be worth commenting in case someone later wonders where the magic number came from, e.g.

```suggestion
            // 128 minus 3px padding on each side
            pixel_size = 122
```